sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -231,6 +231,10 @@ case class StreamingDataSourceV2ScanRelation(
   override protected def stringArgs: Iterator[Any] = stringArgsVal.iterator
 }
 
+object DataSourceV2Table {
```

Comment:
nit: Since it extracts v2 table from the relation, i would prefer it to be `ExtractV2Table`, which makes it explict that it is just an extractor and not a top level class (like `org.apache.spark.sql.catalog.Table`), and follows other clases like `ExtractEquiJoinKeys `, `ExtractJoinWithBuckets ` etc